### ISSUE TITLE
Calling destroy() with criteria on a model without a PK defined generates an error on MySQL

### ISSUE THREAD

**Issue by dcaravana** [balderdashy/waterline-sequel#23]
Using `sails@0.10.5` and `sails-mysql@0.10.11`.

This example:
```
Timeseries.destroy({ response_id: response_id, metric_id: metric_id }).exec(...
```
generates the following error:
```
Error: ER_BAD_FIELD_ERROR: Unknown column 'timeseries.undefined' in 'order clause'
```

The bug is in `WhereBuilder.prototype.single` and it's very easy to spot

```
...
  if(!hop(queryObject, 'sort')) {
    var childPK;

    _.keys(this.schema[this.currentTable].attributes).forEach(function(attr) {
      var expandedAttr = self.schema[self.currentTable].attributes[attr];
      if(!hop(expandedAttr, 'primaryKey')) return;
      childPK = expandedAttr.columnName || attr;
    });

    queryObject.sort = {};
    queryObject.sort[childPK] = -1;
  }
...
```
last two code lines become:
```
    if(childPK) {
      queryObject.sort = {};
      queryObject.sort[childPK] = -1;
    }
```
The workaround of adding explicit sorting to criteria doesn't work either, but helps debug since the syntax of the generated SQL statement is not accepted by MySQL:
```
DELETE `timeseries` 
FROM `TimeSeriesData` AS `timeseries`
WHERE `timeseries`.`response_id` = "2" AND `timeseries`.`metric_id` = "1"
ORDER BY `timeseries`.`response_id` ASC;
```
instead it should be:
```
DELETE FROM `TimeSeriesData`
WHERE `TimeSeriesData`.`response_id` = "2" AND `TimeSeriesData`.`metric_id` = "1"
ORDER BY `TimeSeriesData`.`response_id` ASC;
```
Resorting to a SQL statement through `query()` for now.

**Comment by CWyrtzen** [comment]
@dcaravana was this ever resolved?

**Comment by vkopitsa** [comment]
+1

**Comment by devinivy** [comment]
PR posted: #57.

**Comment by yisyang** [comment]
:+1: @devinivy 
Until that is merged, defining an attribute with primaryKey: false will work as a temporary fix.

**Comment by sailsbot** [comment]
Thanks for posting, @dcaravana.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Reopening until #57 is merged.

**Comment by dcaravana** [comment]
Fix not verified (and still on sails 0.10.5), let me know if we should.

Thanks everyone.

**Comment by sailsbot** [comment]
Thanks for posting, @dcaravana.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Bug with PR pending– reopening.
